core/trino-main/src/test/java/io/trino/type/AbstractTestType.java

```diff
@@ -249,285 +240,141 @@ private void assertPositionValue(Block block, int position, Object expectedStack
 
         if (type.getJavaType() == boolean.class) {
             assertEquals(type.getBoolean(block, position), expectedStackValue);
-            try {
-                type.getLong(block, position);
-                fail("Expected IllegalStateException or UnsupportedOperationException");
-            }
-            catch (IllegalStateException | UnsupportedOperationException expected) {
-            }
-            try {
-                type.getDouble(block, position);
-                fail("Expected IllegalStateException or UnsupportedOperationException");
-            }
-            catch (IllegalStateException | UnsupportedOperationException expected) {
-            }
-            try {
-                type.getObject(block, position);
-                fail("Expected IllegalStateException or UnsupportedOperationException");
-            }
-            catch (IllegalStateException | UnsupportedOperationException expected) {
-            }
+            assertThatThrownBy(() -> type.getLong(block, position)).isInstanceOf(UnsupportedOperationException.class);
+            assertThatThrownBy(() -> type.getDouble(block, position)).isInstanceOf(UnsupportedOperationException.class);
+            assertThatThrownBy(() -> type.getObject(block, position)).isInstanceOf(UnsupportedOperationException.class);
         }
```

Comment:
here IllegalStateException | UnsupportedOperationException has been changed to UnsupportedOperationException, could you pleas ecomment why?